### ISSUE TITLE
Stop covering `tests/dummy/app` files.

### ISSUE THREAD

**Issue by rwjblue** [kategengler/ember-cli-code-coverage#71]
It seems pretty odd that we would want to consider `tests/dummy/app` files as part of code coverage (since they are actually part of the test harness itself).

Does anyone have objections?

/cc @sglanzer @kategengler @rwwagner90 @sandersky @kellyselden

**Comment by rwjblue** [comment]
FWIW, this will require the changes discussed in https://github.com/kategengler/ember-cli-code-coverage/issues/69 to implement (though not particularly hard).

**Comment by kellyselden** [comment]
If you have a complicated acceptance test setup with a large dummy app, and test coverage tells you that some of that dummy app is not being run, would that be something worth preserving?

**Comment by rwjblue** [comment]
> If you have a complicated acceptance test setup with a large dummy app, and test coverage tells you that some of that dummy app is not being run, would that be something worth preserving?

Ya, I can definitely see this argument.

To me it comes down to what the point of "code coverage" is. IMO, the goal is to ensure that the code we are shipping to consumers (`addon/` and `app/` for addons, and `app/` for apps) is sufficiently covered. 

Also, there are a large number of addons that primarily use `tests/dummy/app` for their demo / marketing sites. Much of that code is presentational and may or may not be tested...

**Comment by kellyselden** [comment]
Would it be too complex to not cover "dummy" by default, but allow a config option to turn it on?

**Comment by rwwagner90** [comment]
@kellyselden I approve of disabling by default and allowing an option to turn it on. Would not be hard to do at all.

**Comment by rwjblue** [comment]
Unsure, but we should investigate when we implement.  I agree having finer grained control of "should cover" or "should not cover" would be great....

**Comment by sglanzer** [comment]
I've used an `excludes` filter that was configurable for something like this before, I agree with not including the dummy by default for the same reasons @rwjblue already listed

**Comment by kellyselden** [comment]
I also agree that not covering `dummy` by default is a safe choice.

**Comment by rwwagner90** [comment]
If everyone agrees, I will remove dummy by default and add a config flag to enable dummy coverage.

**Comment by rwjblue** [comment]
I agree that we should remove it from being included by default, but I would prefer to not add a flag specifically for `includeDummy` to config.

I would like to just go whole hog and add a `shouldIncludeFile` hook to config.

Might look like:

```js
  module.exports = {
    shouldIncludeFile: function(project, relativePath, includedByDefault) {
      return; // do the default behavior

      return false || true;
    }
  }
```

This gives us all the most configuration without adding flags for all possible permutations...

**Comment by rwwagner90** [comment]
@adamjmcgrath should we add optional support to cover the `dummy` directory as well?

**Comment by adamjmcgrath** [comment]
> should we add optional support to cover the dummy directory as well?

Personally, I agree with https://github.com/kategengler/ember-cli-code-coverage/issues/71#issuecomment-258902990, so wouldn't use it. Adding optional support would also be complicated.

**Comment by rwwagner90** [comment]
@adamjmcgrath couldn't we just pass a flag and include the directory?

**Comment by adamjmcgrath** [comment]
> @adamjmcgrath couldn't we just pass a flag and include the directory?

That's true, the `includeDummy` option would be fairly straightforward. Sorry, it was the `shouldIncludeFile` solution that I thought would be tricky - because you would not only need to tell Istanbul what _module_ to include, you'd also need to tell it how to find that module on the file system when it's building the report.

**Comment by rwwagner90** [comment]
@adamjmcgrath an `includeDummy` flag would be great! I personally don't think the dummy directory should need to be covered, but if someone wants to include it, no reason we shouldn't support it.
